### ISSUE TITLE
It seems amecs is messing with ctrl and shift keys - while you are pressing them you can't interact with world

### ISSUE THREAD

**Issue by GameBurrow** [Siphalor/amecs#77]
We recently updated our modpack (https://gitlab.com/thesociety/transportcraft-lite/-/releases/2.0.2) to MC 1.20.1 (from 1.19.2) and it seems when amecs is installed we can't use mouse clicks to interact with world while holding ctrl or shift key down. 

Removing AMECS fixes this issue

Some things I also tried
- I tried reset keys to make sure nothing strange carried over from 1.19
- I tried to delete options options.amecsapi.txt
- I tried to downgrade the mod (to older 1.20.1 version)

No idea what to try here anymore as this mod is really useful.

**Comment by Pop000100** [comment]
I have found the conflicting mod in my list. [Useless Reptiles](https://modrinth.com/mod/useless-reptile/version/0.6.3-1.20.1)
It doesn't happen if I extract the embedded library and install them manually, just if the main mod is installed. 
But the mod doesn't have the [github](https://github.com/NordAct/useless-reptile) issues open.

**Comment by GameBurrow** [comment]
> I have found the conflicting mod in my list. [Useless Reptiles](https://modrinth.com/mod/useless-reptile/version/0.6.3-1.20.1) It doesn't happen if I extract the embedded library and install them manually, just if the main mod is installed. But the mod doesn't have the [github](https://github.com/NordAct/useless-reptile) issues open.

Our modpack didn't have that mod in the list so that's not the case for us.

**Comment by Pop000100** [comment]
Yeah, I compared mine to your installed all we had in common. then add a couple more at a time.

**Comment by Pop000100** [comment]
From the Author of Useless Reptiles.

> Sounds more like amecs issue. At worst you can bind fly down key manually. And I don't have issues open on GitHub because I wanted to concentrate all of feedback at Discord. Guess it backfired

> But if you still wonder what could happen: that's the code responsible for reading inputs
https://github.com/NordAct/useless-reptile/blob/1.20.1/src%2Fmain%2Fjava%2Fnordmods%2Fuselessreptile%2Fcommon%2Fentity%2Fbase%2FURRideableDragonEntity.java#L95

> As you can see: all I do is read inputs. If fly down key is unbound, game will use sprint key for it instead (since sprinting when flying doesn't do practically anything anyway) (edited)


Unbinding fly down from shift+c fixes this issue.

**Comment by hatkidchan** [comment]
@Pop000100, did you find the conflicting mod by removing them and testing if it works, or there's a better way to find key conflicts? I have quite a lot of mods to check through (more than 100) and even bisecting would take a while to find the conflict, so I wonder if there's something better than that. Or is there any alternative to AMECS?

**Comment by rocascent** [comment]
Same issue. I need to press `ALT + Shift` instead of `Shift`. 

--edit--
I removed the `shift+*` keybindings from jei and it works fine now.

**Comment by FoilGear** [comment]
I think this is related to this issue I'm experiencing;
My mod selection is stripped down to just Amecs. (i am playing in 1.21.11, fabric 0.19.2)

When I am holding the forward (W) key, while sprinting (Ctrl), and mind you I am _**keeping it held**_, and then I tap (press and release once quickly) Crouch (Shift), it acts as though I let go of Forward (W), and halts all my forward movement, until I release and press Forward (W) again. ; which is really comfortable when I'm in the water and being pursued.

My control scheme usually swaps Shift and Ctrl for my familiarity, but I tried with default binds to make sure it wasn't related; its not, the behavior happens regardless of key configuration.

While I can ultimately learn to let go of Sprint when I'm already Sprinting (tapping it); in the water this is unreliable compared to Holding Sprint to ensure momentum regardless of terrain.
This behavior really hurts and makes the game a little less playable for me, I like this mod for letting me rebind chunk grid and hitboxes with modifier keys.
Hope this is looked into.
